addons/mail/static/src/components/thread_view/thread_view_tests.js

```diff
@@ -1816,6 +1816,190 @@ QUnit.test('composer should be focused automatically after clicking on the send
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ThreadView_alertLoadingFailed',
+        "should show loading error message"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ThreadView_alertLoadingFailedBtn',
+        "should show loading error message button"
+    );
+    assert.containsNone(
+        document.body,
+        '.o_MessageList_loadMore',
+        "should not show load more buttton"
+    );
+});
+
+QUnit.test('Retry loading more messages on failed load more messages should load more messages', async function (assert) {
+    assert.expect(2);
+
+    let calls = 0;
+    this.data['mail.channel'].records.push({
+        channel_type: 'channel',
+        id: 20,
+        is_pinned: true,
+        name: "General",
+    });
+    this.data['mail.message'].records = [...Array(90).keys()].map(id => {
+        return {
+            body: 'coucou',
+            channel_ids: [20],
+            id,
+        };
+    });
+    await this.start({
+        async mockRPC(route, args) {
+            // first call needs to be successful as it is the initial loading of messages
+            // second call comes from load more and needs to fail in order to show the error alert
+            // any later call should work so that retry button and load more clicks would now work
+            if (args.method === 'message_fetch') {
+                calls++;
+                if (calls === 2) {
+                    throw new Error();
+                }
+            }
+            return this._super(...arguments);
+        },
+    });
+    const thread = this.env.models['mail.thread'].findFromIdentifyingData({
+        id: 20,
+        model: 'mail.channel'
+    });
+    const threadViewer = this.env.models['mail.thread_viewer'].create({
+        hasThreadView: true,
+        thread: [['link', thread]],
+    });
+    await this.createThreadViewComponent(threadViewer.threadView, { hasComposer: true });
+    assert.containsN(
+        document.body,
+        '.o_Message',
+        30,
+        "should initially show 30 messages"
+    );
+    // this call will fail
+    await afterNextRender(() => document.querySelector('.o_MessageList_loadMore').click());
+
+    await this.afterEvent({
+        eventName: 'o-thread-view-hint-processed',
+        func: () => document.querySelector('.o_ThreadView_alertLoadingFailedBtn').click(),
+        message: "should wait until channel 20 loaded more messages after clicked on load more",
+        predicate: ({ hint, threadViewer }) => {
+            return (
+                hint.type === 'more-messages-loaded' &&
+                threadViewer.thread.model === 'mail.channel' &&
+                threadViewer.thread.id === 20
+            );
+        },
+    });
+    assert.containsN(
+        document.body,
+        '.o_Message',
+        60,
+        "should show 60 messages as load more has been successfully retried"
+    );
```

Comment:
The 2 assertions `containsN` on `.o_Message` are irrelevant: the assertion that loading failed attempt is successful is made by the `afterEvent` itself, and covers already the intent of test.